examples/3d/parallax_mapping.rs

```diff
@@ -0,0 +1,180 @@
+use bevy::{prelude::*, render::render_resource::TextureFormat, window::close_on_esc};
+
+// A simple 3d scene with a spinning cube with a normal map and height map.
+// Press left mouse button to cycle through different views.
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .insert_resource(Normal(None))
+        .add_startup_system(setup)
+        .add_system(spin)
+        .add_system(update_normal)
+        .add_system(move_camera)
+        .add_system(close_on_esc)
+        .run();
+}
+
+/// SPIN THE CUBE, SPIN IT! SPINNNN!
+#[derive(Component)]
+struct Spin;
+
+/// The camera, used to move camera on click.
+#[derive(Component)]
+struct CameraController;
+
+fn spin(time: Res<Time>, mut query: Query<&mut Transform, With<Spin>>) {
+    for mut transform in query.iter_mut() {
+        transform.rotate_local_y(0.3 * time.delta_seconds());
+        transform.rotate_local_x(0.3 * time.delta_seconds());
+        transform.rotate_local_z(0.3 * time.delta_seconds());
+    }
+}
+
+// Camera positions to cycle through when left-clickig.
+const CAMERA_POSITIONS: &[Transform] = &[
+    Transform {
+        translation: Vec3::new(1.5, 1.5, 1.5),
+        rotation: Quat::from_xyzw(-0.279, 0.364, 0.115, 0.880),
+        scale: Vec3::ONE,
+    },
+    Transform {
+        translation: Vec3::new(2.4, 0.0, 0.2),
+        rotation: Quat::from_xyzw(0.094, 0.676, 0.116, 0.721),
+        scale: Vec3::ONE,
+    },
+    Transform {
+        translation: Vec3::new(2.4, 2.6, -4.3),
+        rotation: Quat::from_xyzw(0.170, 0.908, 0.308, 0.225),
+        scale: Vec3::ONE,
+    },
+    Transform {
+        translation: Vec3::new(-1.0, 0.8, -1.2),
+        rotation: Quat::from_xyzw(-0.004, 0.909, 0.247, -0.335),
+        scale: Vec3::ONE,
+    },
+];
+
+fn move_camera(
+    mut camera: Query<&mut Transform, With<CameraController>>,
+    mut current_view: Local<usize>,
+    button: Res<Input<MouseButton>>,
+) {
+    let mut camera = camera.single_mut();
+    if button.just_pressed(MouseButton::Left) {
+        *current_view = (*current_view + 1) % CAMERA_POSITIONS.len();
+    }
+    let target = CAMERA_POSITIONS[*current_view];
+    camera.translation = camera.translation.lerp(target.translation, 0.2);
+    camera.rotation = camera.rotation.slerp(target.rotation, 0.2);
+}
+
+fn setup(
+    mut commands: Commands,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut normal: ResMut<Normal>,
+    assets: Res<AssetServer>,
+) {
+    let normal_handle = assets.load("textures/parallax_example/paramap_normal.png");
```

Comment:
TODO:
- [ ] rename file to `parallax_mapping_normal.png`
- [ ] re-generate the file so that the normals are not broken (in GIMP, the "flip X axis" should be toggled)
- [ ] Explain how to generate the files, and the fact that the height map is inverted